crates/typst/src/geom/color.rs

```diff
@@ -27,68 +43,429 @@ use crate::syntax::Spanned;
+            Self::LinearRgb(color) => Self::LinearRgb(color.with_alpha(alpha)),
+            Self::Rgba(color) => Self::Rgba(color.with_alpha(alpha)),
+            Self::Cmyk(color) => Self::Cmyk(color.with_alpha(alpha)),
+            Self::Hsl(color) => Self::Hsl(color.with_alpha(alpha)),
+            Self::Hsv(color) => Self::Hsv(color.with_alpha(alpha)),
+        }
+    }
+
+    fn to_array(self, alpha: bool) -> Array {
         match self {
-            Self::Luma(luma) => luma.to_rgba(),
-            Self::Rgba(rgba) => rgba,
-            Self::Cmyk(cmyk) => cmyk.to_rgba(),
+            Color::Luma(color) => color.to_array(alpha),
+            Color::Oklab(color) => color.to_array(alpha),
+            Color::LinearRgb(color) => color.to_array(alpha),
+            Color::Rgba(color) => color.to_array(alpha),
+            Color::Cmyk(color) => color.to_array(alpha),
+            Color::Hsl(color) => color.to_array(alpha),
+            Color::Hsv(color) => color.to_array(alpha),
         }
     }
 }
 
 #[scope]
 impl Color {
-    pub const BLACK: Self = Self::Rgba(RgbaColor::new(0x00, 0x00, 0x00, 0xFF));
-    pub const GRAY: Self = Self::Rgba(RgbaColor::new(0xAA, 0xAA, 0xAA, 0xFF));
-    pub const SILVER: Self = Self::Rgba(RgbaColor::new(0xDD, 0xDD, 0xDD, 0xFF));
-    pub const WHITE: Self = Self::Rgba(RgbaColor::new(0xFF, 0xFF, 0xFF, 0xFF));
-    pub const NAVY: Self = Self::Rgba(RgbaColor::new(0x00, 0x1f, 0x3f, 0xFF));
-    pub const BLUE: Self = Self::Rgba(RgbaColor::new(0x00, 0x74, 0xD9, 0xFF));
-    pub const AQUA: Self = Self::Rgba(RgbaColor::new(0x7F, 0xDB, 0xFF, 0xFF));
-    pub const TEAL: Self = Self::Rgba(RgbaColor::new(0x39, 0xCC, 0xCC, 0xFF));
-    pub const EASTERN: Self = Self::Rgba(RgbaColor::new(0x23, 0x9D, 0xAD, 0xFF));
-    pub const PURPLE: Self = Self::Rgba(RgbaColor::new(0xB1, 0x0D, 0xC9, 0xFF));
-    pub const FUCHSIA: Self = Self::Rgba(RgbaColor::new(0xF0, 0x12, 0xBE, 0xFF));
-    pub const MAROON: Self = Self::Rgba(RgbaColor::new(0x85, 0x14, 0x4b, 0xFF));
-    pub const RED: Self = Self::Rgba(RgbaColor::new(0xFF, 0x41, 0x36, 0xFF));
-    pub const ORANGE: Self = Self::Rgba(RgbaColor::new(0xFF, 0x85, 0x1B, 0xFF));
-    pub const YELLOW: Self = Self::Rgba(RgbaColor::new(0xFF, 0xDC, 0x00, 0xFF));
-    pub const OLIVE: Self = Self::Rgba(RgbaColor::new(0x3D, 0x99, 0x70, 0xFF));
-    pub const GREEN: Self = Self::Rgba(RgbaColor::new(0x2E, 0xCC, 0x40, 0xFF));
-    pub const LIME: Self = Self::Rgba(RgbaColor::new(0x01, 0xFF, 0x70, 0xFF));
+    pub const BLACK: Self = Self::Luma(LumaColor(F32Scalar(0.0)));
+    pub const GRAY: Self = Self::Luma(LumaColor(F32Scalar(0.6666666)));
+    pub const WHITE: Self = Self::Luma(LumaColor(F32Scalar(1.0)));
+    pub const SILVER: Self = Self::Luma(LumaColor(F32Scalar(0.8666667)));
+    pub const NAVY: Self = Self::Rgba(RgbaColor::new(0.0, 0.121569, 0.247059, 1.0));
+    pub const BLUE: Self = Self::Rgba(RgbaColor::new(0.0, 0.454902, 0.85098, 1.0));
```

Comment:
no float ops in const fn :(

Reply:
Yes that's why I had to manually translate them all, and why I had the variant with u8 for RgbaColor 😭